### ISSUE TITLE
Allow user signup using GIthub

### ISSUE THREAD

**Issue by Kajol-Kumari** [anitab-org/vms#1027]
## Description
Currently user can only create a new account using his/her emailID and have to fill all the details manually.

### Update
- [ ] Allow users to signup using gitHub.

- [ ] Collect all the basic user details from github only.

- [ ] Show user a dialog box after signup through Github having a message telling them to complete their profile in order to proceed.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
48 hours

**Comment by Devanshi-123** [comment]
Can i work on this?

**Comment by Kajol-Kumari** [comment]
@Devanshi-123 It's good to see that you are interested in taking up this issue. But all the issues with the label of open source hack will be assigned to the OSH participants starting from 1st of september on the basis of First come first serve so please wait patiently till 1st of september.

**Comment by Devanshi-123** [comment]
Okk mam

**Comment by SanketDG** [comment]
Do you mean to integrate oAuth here? @Kajol-Kumari

**Comment by Kajol-Kumari** [comment]
@SanketDG yes, but i didn't mentioned it explicitly as i want assignee to do a bit of research on finding out the best way to implement it :)

**Comment by Kajol-Kumari** [comment]
Hey Everybody, all who are interested in this issue. Please accept my apologies but according to the new guidelines we received, we will count `First come first serve` on the basis of comments/claims done 1st september onwards. So, everyone who are interested in this issue, do come again on 1st and comment first to get assigned.

**Comment by Nibba2018** [comment]
Hello @Kajol-Kumari , Can I work on this issue?

**Comment by Kajol-Kumari** [comment]
@Nibba2018 please go ahead with this issue.

**Comment by Nibba2018** [comment]
Thank you @Kajol-Kumari

**Comment by satya7289** [comment]
@Kajol-Kumari, It'll be great if you update the issue with allowing a user to sign in/signup using google, and Facebook as well.

**Comment by Kajol-Kumari** [comment]
@satya7289 I will be opening single issue for each of them as keeping everything in single issue will led to the huge PR.

**Comment by kahanikaar** [comment]
Do assign the issue to me if available.

**Comment by Kajol-Kumari** [comment]
Hey @kahanikaar You can go ahead with this. Assigning this to you.

**Comment by Kajol-Kumari** [comment]
@Nibba2018 Can you please provide the update here?

**Comment by Nibba2018** [comment]
@Kajol-Kumari Everything else implemented except for the auto-fill part. #1082

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari Everything else is implemented except for the auto-fill part. #1082

Can you please implement it ASAP and change the PR from `Draft` to `Ready to Review`?

**Comment by Nibba2018** [comment]
@Kajol-Kumari we discussed this issue on Zulip and you decided that you will look into it as I was not able to implement it correctly.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari we discussed this issue on Zulip and you decided that you will look into it as I was not able to implement it correctly.

Ah, probably I missed it, btw If I am not wrong I had tagged @satya7289 to help you out with this. Btw I would like to see what you have implemented till now. Can you please share a GIF in the PR description?

**Comment by Nibba2018** [comment]
@Kajol-Kumari Done
